### ISSUE TITLE
localization (nativeLang.xml) should allow numeric XML entities

### ISSUE THREAD

**Issue by pryrt** [notepad-plus-plus/notepad-plus-plus#13379]
### Description of the Issue

I was trying to replace the literal `⇅` and `⤵` and `⤴` with XML numeric entities, because some fonts don't have the fancy arrow characters , so users/translators were getting blank boxes, per [this forum discussion](https://community.notepad-plus-plus.org/post/84945).  See this example, where `⇅` shows okay, but not the other two:

![image](https://user-images.githubusercontent.com/17455758/226211435-318aebf3-02e1-49cc-a42c-e160abde7c58.png)

But when I tried using XML's numeric entities, like use `&#x21C5;` or `&#8645;` for `⇅` , it does not get interpreted as an entity, and instead stays as the literal sequence.  Even `&#34;` for a quote-mark `"` doesn't work, even though `&quot;` _does_ work:

![image](https://user-images.githubusercontent.com/17455758/226211290-22e4fd71-6d14-417e-9389-3e32e613ea45.png)

It confuses me (and translators and users) that some entities do work, but other valid XML entities do not.  All valid XML entities should work if any work.

I cannot put in a PR per [that forum post](https://community.notepad-plus-plus.org/post/84945) until the XML numeric entity handling in translation strings works reliably for all decimal and hexadecimal entities.

### Steps to Reproduce the Issue

1. Edit `nativeLang.xml` to include an entity, such as 
    ```
				<Menu>
					<Item id="1726" name="| &#x21C5; &quot;Suchen&#34; &#x22;mit Ersetzen vertauschen"/>
					<Item id="1727" name="| &#10549; Kopiere von Suchen nach Ersetzen"/>
					<Item id="1728" name="| &#10548; Kopiere von Ersetzen nach Suchen"/>
				</Menu>
    ```
2. Save `nativeLang.xml`
2. Restart Notepad++
3. Open up the GUI interface that uses that text
4. See ![image](https://user-images.githubusercontent.com/17455758/226211290-22e4fd71-6d14-417e-9389-3e32e613ea45.png)
5. Neither the `&#x21C5;` hex-style numeric XML entity, nor the `&#10549;` decimal-style numeric XML entity works.  The `&#x22;` hex entity does work for quote, as does the named `&quot;` entity, but the `&#34;` decimal XML entity does not work for quot

### Expected Behavior

All valid XML entities (which include the main 5 named entities, plus any of the `&#`-prefixed hex or decimal entities, should be interepreted as the proper character for any translatable string from the nativeLang.xml and its sources from the `localization/` directory.

### Actual Behavior

Some (probably most) numeric entities do not work.

### Debug Information

```
Notepad++ v8.5.1   (64-bit)
Build time : Mar 17 2023 - 02:35:00
Path : C:\usr\local\apps\npp\npp.8.5.1-RC1.portable.x64\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 11 Home (64-bit)
OS Version : 22H2
OS Build : 22621.1413
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
```

**Comment by dinkumoil** [comment]
This is no bug, your input data is not correct.

Translation files are UTF-8 encoded, thus you have to provide the hex codes of the UTF-8 code points for the characters. Try the following:

`⇅` -> `&#xE2;&#x87;&#x85;`
`⤵` -> `&#xE2;&#xA4;&#xB5;`
`⤴` -> `&#xE2;&#xA4;&#xB4;`

**Comment by ArkadiuszMichalski** [comment]
@pryrt 

Have you got cloned this repo https://github.com/notepad-plus-plus/notepad-plus-plus.wiki.git ? By accident, instead of adding a new entry, I changed the name of the page and all its previous history was lost. If you have and can restore it by push then push the latest version.

If it can be restored, I will add changes for 8.4.9 and 8.5 again. And if not, I will change the name of the page to what it was (probably just Changelog or Changes?).

**Comment by ozone10** [comment]
@pryrt 
this should be fixed thanks to replacement to pugixml. Could you test it again?

**Comment by pryrt** [comment]
Yes, with pugixml, the example from above works, so all the variants of the numeric entities are working correctly.
